user/configuration.rst

```diff
@@ -0,0 +1,114 @@
+* ``dependencies`` (list of strings) to declare additional dependencies on other packages.
+  For more fine grain control it is also possible to set ``build-dependencies``, ``run-dependencies``, and ``test-dependencies``.
+* ``hooks`` (list of relative paths within the install prefix) to declare additional scripts to be sourced.
+* Any command line argument.
+  The leading single or double dash must be omitted.
+
+Values for command line arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The value type depends on the kind of command line argument:
+
+* For flags which are not followed by a value the value can be either ``true`` or ``false``.
+* For options followed by a single decimal / float the value must be a decimal / float.
+* For options followed by a single value the value must be a string.
+* For options followed by one or more values the value must be a list where each item can be any of the mentioned types.
+
+An example declaring an environment hook which should be sourced for a package:
+
+.. code-block:: yaml
+
+    {
+        "hooks": ["share/pkgname/hook/something.sh"]
+    }
+
+.meta files
+-----------
+
+The first level of the configuration file is a dictionary.
+The only two supported keys are:
+* ``names`` to provide settings based on the package name.
+* ``paths`` to provide settings based on the package path.
+
+By package name
+~~~~~~~~~~~~~~~
+
+.. note::
+
+    Providing metadata based on the package name only works if the package can be identified and the name can be determined.
+    Otherwise using a `colcon.pkg` file or the `By package path`_ configuration is necessary.
+
+The value under the ``names`` key is again a dictionary.
+
+The key is the name of the package.
+The value can contain the same package specific settings as described in the `colcon.pkg files`_ section above.
+The only except is that specifying a package name is not supported.
+
+By package path
+~~~~~~~~~~~~~~~
+
+The value under the ``paths`` key is again a dictionary.
+
+The key is the path of the package.
+It can be either absolute or relative to the ``.meta`` file.
+The value can contain the same package specific settings as described in the `colcon.pkg files`_ section above.
+This can be used if the package name can't be determined automatically and placing a ``colcon.pkg`` file into the package directory is undesired.
+
+Package specific configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The package specific part under the package name or package path has the same content as the package specific configuration files described in the `colcon.pkg files`_ section above.
+
+Using .meta files
+~~~~~~~~~~~~~~~~~
+
+Some configuration files are being picked up by default.
+The following are a few examples (see e.g. ``colcon build --help``):
+
+* When ``--ignore-user-meta`` is not passed any file ending with ``.meta`` in any recursive subdirectory of ``$COLCON_HOME/metadata`` is being used.
```

Comment:
Is $COLCON_HOME initialized to something?  Is that mentioned somewhere else in the docs? It may be worth repeating here.

Reply:
Maybe just a note in place: see 48a7b550784cb01eb8fa79a94951b5c1b6d385ed